MSAL/test/app/ios/MSALTestAppB2CAuthorityViewController.m

```diff
@@ -0,0 +1,71 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSALTestAppB2CAuthorityViewController.h"
+#import "MSALTestAppSettings.h"
+
+@interface MSALTestAppB2CAuthorityViewController ()
+
+@end
+
+@implementation MSALTestAppB2CAuthorityViewController
+
++ (instancetype)sharedController
+{
+    static MSALTestAppB2CAuthorityViewController *s_controller = nil;
+    static dispatch_once_t once;
+    
+    dispatch_once(&once, ^{
+        s_controller = [[MSALTestAppB2CAuthorityViewController alloc] init];
+    });
+    
+    return s_controller;
+}
+
+- (instancetype)init
+{
+    return [super initWithAuthorities:[[MSALTestAppSettings b2cAuthorities] mutableCopy]
+                      addUserDefaults:[[NSUserDefaults standardUserDefaults] objectForKey:@"saved_b2cAuthorities"]];
+}
+
+- (void)saveAuthority:(NSString *)authority
+{
+    if(![_savedAuthorities containsObject:authority])
+    {
+        [_savedAuthorities addObject:authority];
+    }
+    
+    [[NSUserDefaults standardUserDefaults] setObject:_savedAuthorities forKey:@"saved_b2cAuthorities"];
+    if(![_authorities containsObject:authority])
```

Comment:
nit: technically, you can always only check that authorities doesn't contain authority, because authorities would have savedAuthorities in it. Also, we should first check if authorities already has it, because otherwise we'll add it to savedAuthorities, although it might have already been present in authorities.

Reply:
@ I think the order here does not matter and both checks need to be present. _savedAuthorities only takes care of the authorities added using the add button by saving them in the userdefaults. We also add the same authorities to the _authorities to make sure the table view is updated as its data source points to the _authorities. Then in the init method we retrieve the user defaults in the _savedAuthorities and for all _saved authorities not present in the _authorities , we add them to the _authorities. Please correct me if I am wrong.